### ISSUE TITLE
`max_connections` threshold strategies for concurrent multi-database access

### ISSUE THREAD

**Issue by adamhamlin** [brianc/node-postgres#3081]
Obviously the pool is great when you're trying to limit concurrent connections to a single database. But in the scenario where your app is simultaneously reaching out to multiple databases on the same postgres server, it's not too hard to blow past the server's `max_connections` limit.

Any interest in a PR that would create a mechanism to throttle total clients across pools for a given host? Usage might be something like the following:

```ts
const serverPools = new pg.ServerPools();
const pool = serverPools.getOrCreate(connInfoForSpecificDb);
await pool.query('SELECT NOW()');
```

Obviously, there are lots of questions/details surrounding things like how you might throttle and _fairly_, but just curious what you think of the idea @brianc .

Thanks.

**Comment by alpharder** [comment]
@adamhamlin Is creating separate pg pool per used database not an option for you?

**Comment by alpharder** [comment]
@adamhamlin yeah, I see... 

The only thing that comes to my mind is PGBouncer

**Comment by adamhamlin** [comment]
Yeah, potentially. But that's a much bigger lift and involves other parts of the stack.

I'm mostly interested in an application code solution. Of course, these limits would only be enforced for a single `pg.ServerPools` instance--but that's no different than how `pg.Pool` works today.

**Comment by charmander** [comment]
I think you should create a package for this!

**Comment by adamhamlin** [comment]
@charmander thanks, yeah I think it could be useful. Tho I don't know if it can be done "right" as a new package vs. an addition to `pg.Pool` with access to all the pool internals.

**Comment by charmander** [comment]
It can.

**Comment by adamhamlin** [comment]
I think a "fair" paradigm when approaching `max_connections` might involve temporarily lowering the number of connections that a single database pool can have to allow for more pools to be created/utilized. That's the kind of thing I think one might want (need?) access to the internals for
